apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/domains/page-client.tsx

```diff
@@ -30,9 +31,8 @@ function EditDialog(props: {
   }
 )) {
   const domainFormSchema = yup.object({
-    domain: urlSchema
-      .url("Invalid URL")
-      .transform((value) => 'https://' + value)
+    domain: yup.string()
+      .matches(DOMAIN_REGEX, "Invalid domain")
```

Comment:
it's better to the URL methods from stack-shared for this. What I'm thinking is something like this:

```ts
const urlIfValid = createUrlIfValid(`https://${domain}`);
return !!urlIfValid && urlIfValid.domain === domain;
```

Are there any edge cases I'm missing? If not, feel free to add this as a `isValidDomain` utility in stack-shared